llvm/lib/CodeGen/AsmPrinter/AsmPrinter.cpp

```diff
@@ -1800,6 +1804,31 @@ void AsmPrinter::emitFunctionBody() {
         }
 
         emitInstruction(&MI);
+        // Generate labels for function calls so we can record the correct
+        // instruction offset. The conditions for generating the label must be
+        // the same as the ones for generating the stackmap call in
+        // `Transforms/Yk/Stackmaps.cpp`, as otherwise we could end up with
+        // wrong offsets (e.g. we create a label here but the corresponding
+        // stackmap call was ommitted, and this label is then used for the
+        // following stackmap call).
+
+        // Convoluted way of finding our whether this function call is an intrinsic.
+        bool IsIntrinsic = false;
+        if (MI.getNumExplicitDefs() < MI.getNumOperands()) {
+          IsIntrinsic = MI.getOperand(MI.getNumExplicitDefs()).isIntrinsicID();
+        }
+        if (YkStackMapOffsetFix && MI.isCall() && !MI.isInlineAsm() &&
+            (MI.getOpcode() != TargetOpcode::STACKMAP) &&
+            (MI.getOpcode() != TargetOpcode::PATCHPOINT) &&
+            (MI.getOpcode() != TargetOpcode::G_MEMSET) &&
```

Comment:
What's the memset thing about?

Reply:
This is possibly a left over from when we added the memset thingy to for debugging. It can possibly go, since I have the addtional intrinsic check below, which this should be caught by too. Let me try to remove it.

Reply:
Fixed in [e81d2f8](https://github.com/ykjit/ykllvm/pull/52/commits/e81d2f87d0e60a96fd2274aecbceff64fc4a1ee3)

Reply:
What about this one?

Reply:
It's fixed in [e81d2f8](https://github.com/ykjit/ykllvm/pull/52/commits/e81d2f87d0e60a96fd2274aecbceff64fc4a1ee3)

Reply:
Oops, we raced!